src/harness/fourslash.ts

```diff
@@ -1797,6 +1797,10 @@ namespace FourSlash {
                 // this.languageService.getScriptLexicalStructure(fileName);
             }
 
+            if (this.currentCaretPosition < 0) {
+                this.currentCaretPosition = 0;

```

Comment:
If the caret is in the middle of some range that is deleted, should the caret go to the beginning of the range? Or be invalidated?

I'm unsure this can happen in a real editing scenario, but it can make sense for sequencing checks in tests.